File: Regression/simple_linear_regression.R
# simple Linear Regression in R

# Loading dataset
df <- read.csv("Salary_Data.csv")
head(df)

# spliting the data
library(caTools)
split <- sample.split(df$Salary, SplitRatio = 0.8)
train.set <- subset(df, split == TRUE)
test.set <- subset(df, split == FALSE)

# Linear Regression
reg <- lm( Salary ~ YearsExperience , data = df)
summary(reg)

# Predicting on test
pred <- predict(reg, test.set)

#Vizualization
require(ggplot2)
pl <- ggplot() +  
  geom_point(data = train.set, aes(YearsExperience, Salary), color = 'red') + 
  geom_line(aes(x = train.set$YearsExperience, y = predict(reg, train.set)), color = 'blue') 

File: Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sat Feb 24 19:57:08 2018

@author: Rupesh
"""


# Linear Regression

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
plt.style.use("ggplot")
# loading dependies

df = pd.read_csv("Salary_Data.csv")
df.head()
X = df.iloc[:, :-1].values
y = df.iloc[:, 1].values

# Ploting the data 
sns.pairplot(df)
plt.show()
"""
Seems like the is a relation between the data 
on Experience and Salary 
which is obvious
"""


# spliting the dataset into train and test sets
from sklearn.cross_validation import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y)


# appling the Regression model
from sklearn.linear_model import LinearRegression

lm = LinearRegression()
lm.fit(X_train, y_train)


pred = lm.predict(X_test)

plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_test, pred, color = 'blue')
plt.title("Salary V Experience")
plt.xlabel("Experience")
plt.ylabel("Salary")
lm.score(X_test, y_test)
# gives the accuracy at which the model predicts

